ykcapi/src/lib.rs

```diff
@@ -152,10 +162,14 @@ pub extern "C" fn yk_stopgap(
     #[cfg(feature = "yk_jitstate_debug")]
     print_jit_state("enter-stopgap");
 
-    // Parse AOTMap.
-    let aotmap = unsafe { slice::from_raw_parts(aotmap.addr as *const AOTVar, aotmap.length) };
+    // Parse the live AOT values.
+    let aotvalsptr =
+        unsafe { (aotvals.addr as *const u8).offset(isize::try_from(aotvals.offset).unwrap()) };
+    let aotvals = unsafe { slice::from_raw_parts(aotvalsptr as *const AOTVar, aotvals.length) };
 
     // Parse active frames vector.
+    // Note that the memory behind this slice is allocated on the stack (of the compiled trace) and
+    // thus doesn't need to be freed.
```

Comment:
We replace the frame of the trace in the event of guard failure, is that correct?

Reply:
Yes, we are rolling back to the frame that called the trace, so everything on the stack gets nuked.